fs/path_test.go

```diff
@@ -183,6 +183,51 @@ func TestRootPath(t *testing.T) {
 	t.Run("SymlinkEmpty", testRootPathSymlinkEmpty)
 }
 
+func TestDirectoryCompare(t *testing.T) {
+	for i, tc := range []struct {
+		p1 string
+		p2 string
+		r  int
+	}{
+		{"", "", 0},
+		{"", "/", -1},
+		{"/", "", 1},
+		{"/", "/", 0},
+		{"", "", 0},
+		{"/dir1", "/dir1/", -1},
+		{"/dir1", "/dir1", 0},
+		{"/dir1/", "/dir1", 1},
+		{"/dir1", "/dir2", -1},
+		{"/dir2", "/dir1", 1},
+		{"/dir1/1", "/dir1-1", -1},
+		{"/dir1-1", "/dir1/1", 1},
+		{"/dir1/dir2", "/dir1/dir2", 0},
+		{"/dir1/dir2", "/dir1/dir2/", -1},
+		{"/dir1/dir2", "/dir1/dir2/f1", -1},
+		{"/dir1/dir2/", "/dir1/dir2", 1},
+		{"/dir1/dir2/f1", "/dir1/dir2", 1},
+		{"/dir1/dir2-f1", "/dir1/dir2", 1},
+		{"/dir1/dir2-f1", "/dir1/dir2/", 1},
+		{"/dir1/dir2/", "/dir1/dir2/", 0},
+		{"/dir1/dir2你", "/dir1/dir2a", 1},
+		{"/dir1/dir2你", "/dir1/dir2", 1},
+		{"/dir1/dir2你", "/dir1/dir2/", 1},
+		{"/dir1/dir2你/", "/dir1/dir2/", 1},
+		{"/dir1/dir2你/", "/dir1/dir2你/", 0},
+		{"/dir1/dir2你/", "/dir1/dir2你好/", -1},
+		{"/dir1/dir2你/", "/dir1/dir2你-好/", -1},
```

Comment:
@dmcgowan you should try the character `biang` 😄

Reply:
no unicode for biang :P